pkg/util/k8s/event_emmiter.go

```diff
@@ -0,0 +1,37 @@
+package k8s
```

Comment:
Move this to `pkg/eventemitter/eventemitter.go` so we do `eventemitter.New()`

Also are we really sure that we need the interface ? I suspect is for unit testing if this is the case maybe we can mock `GetEventRecorderFor` somehow passing a fake manager.

Reply:
Nah EventEmitter feels right.

Well maybe the most important part is tu split where do we trigger the events and how do we format them, for that I sould say that we create some high level functions at `EventEmitter` that reads the networkaddons config and format the events as needed

```golang
func EmmitAvailable(configName string)
func EmmitProgressing(configName string)
func EmmitFailing(configName string)
func EmmitWarning(configName string, reason string, message string)
```

Reply:
> Let's create some high level functions with event formating so we split responsabilities, added a comment to it.

Thanks @qinqon 
I added the external function. I hope that's what you meant.

I didn't understand what is EmitWarning. can you elaborate?

I agree we need to think together about were we wnat to place the events. this is my suggestion, but I am not really sure of it...